lib/mapObjects/CObjectHandler.h

```diff
@@ -57,7 +58,13 @@ class DLL_LINKAGE IObjectInterface
 	virtual void garrisonDialogClosed(const CGHeroInstance *hero) const;
 	virtual void heroLevelUpDone(const CGHeroInstance *hero) const;
 
-//unified interface, AI helpers
+	//unified helper to show info dialog for object owner
+	virtual void showInfoDialog(const ui32 txtID, const ui16 soundID = 0, EInfoWindowMode mode = EInfoWindowMode::AUTO) const;
+
+	//unified helper to show a specific window
+	static void openWindow(const EOpenWindowMode type, const int id1, const int id2 = -1);
```

Comment:
But there was converted from a bunch of static functions, so, I made it static. I do not think than we should avoid static like plague) I think if we will even need instance access, then we can convert this to non-static.

Reply:
Static does have usage. For example singleton access.

But other than that - hardly. In my opinion static is indication of poor design.
If you have static method it is likely that this method should be either part of some namespace (and not part of class) or should be part of another class.

Whether method makes any access to class members is (normally) an implementation detail and should not be exposed to user code.

> I think if we will even need instance access, then we can convert this to non-static.

Problem is, if/when we do so, it won't be as simple as removing "static" from method declaration. This also means updating every call that used access without instance. And this part might not be trivial. No need to promote poor code that will have to be rewritten later.

Reply:
For now, there is no call of openWindow with class access notation. So, changing it to instance method is zero-cost.

Reply:
static/non-static is more like behavior of instance or class in general. If the method describes instance behavior but still does not use instance state it is still better to leave it instance method. But if it describes something more generic, overall class behavior, some meta-information it may be static. This thing looks like a factory method so it is ok for it to be static. Otherwise we will need some window manager

Reply:
This is not factory method. 
It generates and send network pack that asks client to show window.

And internally it uses another static member (that should not be static)